### DISCUSSION TITLE
How to add Carousel for Related / Recent / Crossell Products ?

### DISCUSSION THREAD

**Discussion by rutulritz** [breezefront/community#35] [1. Help] [answered]
Hi,

I am using breeze theme on my website. 

I want to show carousel for Recent Products, Related Products and Crossell Products. 

How can I implement this over breeze evolution?

Can you please guide me on this. 

Thanks

**Comment by vovayatsyuk** [comment, marked as answer]
You can wrap the products into a `div.breeze-carousel` and breeze will transform all listings into sliders.

In our Argento themes we do this via XML layout update instructions:

```xml
<referenceContainer name="page.bottom.container">
    <container name="before-footer.recent-wrapper" htmlTag="div" htmlClass="recent-products hide show-on-ready" after="-">
        <container name="before-footer.bottom.carousel" htmlTag="div" htmlClass="breeze-carousel container"/>
    </container>
</referenceContainer>
<referenceBlock name="catalog.product.viewed" template="product/widget/viewed/grid.phtml"/>
<move element="catalog.product.viewed" destination="before-footer.bottom.carousel"/>
```

**Reply by rutulritz** [reply]
Ok Thanks, I will try it.

In which file have to add it?

**Reply by rutulritz** [reply]
Thanks @vovayatsyuk , I added Recent & Related Products Carousel on PDP page

Now I tried the same to add Crosssel product carousel on Cart page but it is not working. [Breeze exp is disabled for Cart page]

Please guide me for the same.

Thanks

**Reply by vovayatsyuk** [reply]
Sorry, I can't help you with that. How would you do that in Luma? The same solution should work as well.
